property_tests/tests/null/test/test.ts

```diff
@@ -5,48 +5,73 @@ import { JsFunctionNameArb } from '../../../arbitraries/js_function_name_arb';
 import { TestSample } from '../../../arbitraries/test_sample_arb';
 import { createUniquePrimitiveArb } from '../../../arbitraries/unique_primitive_arb';
 import { getActor, runPropTests } from '../../../../property_tests';
+import { Candid } from '../../../arbitraries/candid';
+import { Test } from '../../../../test';
 
 const NullTestArb = fc
-    .tuple(createUniquePrimitiveArb(JsFunctionNameArb), fc.array(NullArb))
-    .map(([functionName, nulls]): TestSample => {
-        const paramCandidTypes = nulls
+    .tuple(
+        createUniquePrimitiveArb(JsFunctionNameArb),
+        fc.array(NullArb),
+        NullArb
+    )
+    .map(([functionName, paramNulls, returnNull]): TestSample => {
+        const imports = returnNull.src.imports;
+
+        const paramNames = paramNulls.map((_, index) => `param${index}`);
+        const paramCandidTypes = paramNulls
             .map((Null) => Null.src.candidType)
             .join(', ');
-        const returnCandidType = 'Null';
-        const paramNames = nulls.map((_, index) => `param${index}`);
 
-        const areAllNull = paramNames.reduce((acc, paramName) => {
-            return `${acc} && ${paramName} === null`;
-        }, 'true');
+        const returnCandidType = returnNull.src.candidType;
+
+        const body = generateBody(paramNames, returnNull);
 
-        const allNullCheck = `if (!(${areAllNull})) throw new Error("Not all of the values were null")`;
+        const test = generateTest(functionName, paramNulls, returnNull);
 
         return {
             functionName,
-            imports: ['Null'],
+            imports,
             paramCandidTypes,
             returnCandidType,
             paramNames,
-            body: `
-            ${allNullCheck}
-
-            return null;
-        `,
-            test: {
-                name: `test ${functionName}`,
-                test: async () => {
-                    const actor = getActor('./tests/null/test');
-
-                    const result = await actor[functionName](
-                        ...nulls.map((sample) => sample.value)
-                    );
-
-                    return {
-                        Ok: result === null
-                    };
-                }
-            }
+            body,
+            test
         };
     });
 
 runPropTests(NullTestArb);
+
+function generateBody(paramNames: string[], returnNull: Candid<null>): string {
+    const areAllNull = paramNames.reduce((acc, paramName) => {
+        return `${acc} && ${paramName} === null`;
+    }, 'true');
+
+    const allNullCheck = `if (!(${areAllNull})) throw new Error("Not all of the values were null")`;
+
+    return `
+        ${allNullCheck}
+
+        return ${returnNull.value};
```

Comment:
This seems not correct. I think you want the string representation of the null value. Put a todo in so that we don't forget about this. We just want the code to say what we're meaning.